guide/ops/locations/_cloudstack.md

```diff
@@ -0,0 +1,138 @@
+
+The identity is the "api key" and the credential is the "secret key". These can be generated in 
+the cloudstack gui: under accounts, select "view users", then "generate key".
+
+    location:
+      jclouds:cloudstack:
+        endpoint: https://cloud.acme.com/client/api
+        identity: abcdefghijklmnopqrstuvwxyz01234567890-abcdefghijklmnopqrstuvwxyz01234567890-abcdefghij
+        credential: mycred-abcdefghijklmnopqrstuvwxyz01234567890-abcdefghijklmnopqrstuvwxyz01234567890-abc
+
+
+### Common Configuration Options
+
+Below are examples of configuration options that use values specific to CloudStack environments:
+
+* The `imageId` is the template id. For example,
+  `imageId: db0bcce3-9e9e-4a87-a953-2f46b603498f`.
+
+* The `region` is CloudStack zone id.
+  For example `region: 84539b9c-078e-458a-ae26-c3ffc5bb1ec9`..
+
+* `networkName` is the network id (within the given zone) to be used. For example, 
+  `networkName: 961c03d4-9828-4037-9f4d-3dd597f60c4f`.
+
+For further configuration options, consult 
+[jclouds CloudStack template options](https://jclouds.apache.org/reference/javadoc/1.9.x/org/jclouds/cloudstack/compute/options/CloudStackTemplateOptions.html).
+These can be used with the **[templateOptions](#custom-template-options)** configuration option.
+
+
+### Using a Pre-existing Key Pair
+
+The configuration below uses a pre-existing key pair:
+
+    location:
+      jclouds:cloudstack:
+        ...
+        jclouds.openstack-nova.auto-generate-keypairs: false
+        loginUser: root
+        loginUser.privateKeyFile: /path/to/keypair.pem
+        keyPair: my-keypair
+
+
+### Using Pre-existing Security Groups
+
+To specify existing security groups, their ids must be used rather than their names (note this
+differs from the configuration on other clouds!).
+ 
+The configuration below uses a pre-existing security group:
+
+    location:
+      jclouds:cloudstack:
+        ...
+        templateOptions:
+          generateSecurityGroup: false
+          securityGroupIds:
+          - 12345678-90ab-def0-1234-567890abcdef
+
+
+### Using Static NAT
+
+Assigning a public IP to a VM at provision-time is referred to as "static NAT" in CloudStack
+parlance. To give some consistency across different clouds, the configuration option is named
+`autoAssignFloatingIp`. For example, `autoAssignFloatingIp: false`.
+
+
+### Cloudmonkey CLI
+
+The [CloudStack Cloudmonkey CLI](https://cwiki.apache.org/confluence/display/CLOUDSTACK/CloudStack+cloudmonkey+CLI)
+is a very useful tool for validating that credentials are correct, and for querying the API to find the correct
+zone ids etc.
+
+Useful commands include:
+
+    # for finding the ids of the zones:
+    cloudmonkey api listZones
+
+    # for finding the ids of the networks.
+    cloudmonkey api listNetworks | grep -E "id =|name =|========="
+
+### CloudStack Troubleshooting
+
+These troubleshooting tips are more geared towards problems encountered in old test/dev 
+CloudPlatform environment.
```

Comment:
Doh, "CloudPlatform" was the name of Citrix's offering (which was pretty much CloudStack with support, plus a slightly different release cycle for testing etc). I'll change to CloudStack!